src/legacy/core_plugins/kibana/common/field_formats/types/relative_date.ts

```diff
@@ -18,28 +18,33 @@
  */
 
 import moment from 'moment';
+import { FieldFormat, TEXT_CONTEXT_TYPE } from '../../../../../../plugins/data/common/';
 
-export function createRelativeDateFormat(FieldFormat) {
-  return class RelativeDateFormat extends FieldFormat {
-    constructor(params) {
+export function createRelativeDateFormat() {
+  class RelativeDateFormat extends FieldFormat {
+    static id = 'relative_date';
+    static title = 'Relative Date';
+    static fieldType = 'date';
+
+    constructor(params: any) {
       super(params);
     }
 
-    _convert(val) {
-      if (val === null || val === undefined) {
-        return '-';
-      }
+    _convert = {
```

Comment:
I think we should refactor base class (FieldFormat) and remove _convert at all. Right now property is not protected by TypeScript and we can set any value to this property. 

I offer to use the following structure for formatters: 

![image](https://user-images.githubusercontent.com/20072247/66112640-e7b88300-e5d3-11e9-9c2d-cf99d0bf9ca6.png)

@ppisljar  any ideas?

Reply:
++

also can we provide default htmlConvert function (which just calls textConvert())

Reply:
`FieldFormat` and derived classes have been changed.